### ISSUE TITLE
worksheet.add_protected_range() - impossible to create the `warning_only=True` range

### ISSUE THREAD

**Issue by aperepel** [burnash/gspread#1437]
Hi, I wonder if there's a bug.

When using `worksheet.add_protected_range(name="A1:B2", warning_only=True)`, I'm getting the `TypeError: Worksheet.add_protected_range() missing 1 required positional argument: 'editor_users_emails'`

However, adding `editor_users_emails` triggers a 400 Bad Request, as the library seems to always require it:

```
worksheet.add_protected_range(name="A1:B2", warning_only=True, editor_users_emails="me@example.com")

--- (and the error is)

ProtectedRange is warningOnly. Editors cannot be set on it.
```

I think the gspread library must omit the editor node altogether from the request if `warning_only=True`.

**Comment by aperepel** [comment]
FYI, I had to fallback to the raw API, here's what the working request looks like:
```
grid_range = gspread_utils.a1_range_to_grid_range(cell_range)

    request_body = {
        "requests": [
          {
              "addProtectedRange": {
                  "protectedRange": {
                      "range": {
                          "sheetId": int(worksheet_id),
                          "startRowIndex": grid_range['startRowIndex'],
                          "endRowIndex": grid_range['endRowIndex'],
                          "startColumnIndex": grid_range['startColumnIndex'],
                          "endColumnIndex": grid_range['endColumnIndex'],
                      },
                      "warningOnly": True
                  }
              }
          }
       ]
    }

    res = gsheet.batch_update(request_body)
```

**Comment by alifeee** [comment]
Hi, thanks for this issue. You have made it very clear the problem, which is great.

I have made a fix with #1439. If no problems arise, it should appear in the next minor release :)
